flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/sink/config/MongoWriteOptions.java

```diff
@@ -0,0 +1,200 @@
+        return batchSize;
+    }
+
+    public long getBatchIntervalMs() {
+        return batchIntervalMs;
+    }
+
+    public int getMaxRetries() {
+        return maxRetries;
+    }
+
+    public long getRetryIntervalMs() {
+        return retryIntervalMs;
+    }
+
+    public DeliveryGuarantee getDeliveryGuarantee() {
+        return deliveryGuarantee;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        MongoWriteOptions that = (MongoWriteOptions) o;
+        return batchSize == that.batchSize
+                && batchIntervalMs == that.batchIntervalMs
+                && maxRetries == that.maxRetries
+                && retryIntervalMs == that.retryIntervalMs
+                && deliveryGuarantee == that.deliveryGuarantee;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
+                batchSize, batchIntervalMs, maxRetries, retryIntervalMs, deliveryGuarantee);
+    }
+
+    public static MongoWriteOptionsBuilder builder() {
+        return new MongoWriteOptionsBuilder();
+    }
+
+    /** Builder for {@link MongoWriteOptions}. */
+    @PublicEvolving
+    public static class MongoWriteOptionsBuilder {
+        private int batchSize = BUFFER_FLUSH_MAX_ROWS.defaultValue();
+        private long batchIntervalMs = BUFFER_FLUSH_INTERVAL.defaultValue().toMillis();
+        private int maxRetries = SINK_MAX_RETRIES.defaultValue();
+        private long retryIntervalMs = SINK_RETRY_INTERVAL.defaultValue().toMillis();
+        private DeliveryGuarantee deliveryGuarantee = DeliveryGuarantee.AT_LEAST_ONCE;
+
+        private MongoWriteOptionsBuilder() {}
+
+        /**
+         * Sets the maximum number of actions to buffer for each bulk request. You can pass -1 to
+         * disable it. The default flush size is 1000.
+         *
+         * @param batchSize the maximum number of actions to buffer per bulk request.
+         * @return this builder
+         */
+        public MongoWriteOptionsBuilder setBatchSize(int batchSize) {
+            checkArgument(
+                    batchSize == -1 || batchSize > 0,
+                    "Max number of batch size must be larger than 0.");
+            this.batchSize = batchSize;
+            return this;
+        }
+
+        /**
+         * Sets the bulk flush interval, in milliseconds. You can pass -1 to disable it.
+         *
+         * @param batchIntervalMs the batch flush interval, in milliseconds.
+         * @return this builder
+         */
+        public MongoWriteOptionsBuilder setBatchIntervalMs(long batchIntervalMs) {
+            checkArgument(
+                    batchIntervalMs == -1 || batchIntervalMs >= 0,
+                    "The batch flush interval (in milliseconds) between each flush must be larger than "
+                            + "or equal to 0.");
```

Comment:
What does it mean for the interval to be 0 (opposed to -1)?

Reply:
There are currently three ways to write:
1.  flush only on checkpoint
2. flush for for every single element
3. flush when time interval or  batch size exceed limi

| batch interval \\ batch size  |  -1  |  1  | > 1  |
|--|--|--|--|
| -1  |  flush on checkpoint  |  flush for every single element | flush when batch size exceed limit |
| 0  |  flush for every single element  |  flush for every single element  | flush for every single element |
| > 0  |  flush when time interval exceed limit  | flush for every single element | flush when time interval or  batch size exceed limit |

Reply:
We should put this somewhere into the docs.